pkg/endpoint/endpoint.go

```diff
@@ -538,19 +547,12 @@ func (e *Endpoint) DeepCopy() *Endpoint {
 	return cpy
 }
 
-// StringIDLocked returns the endpoint's ID in a string. Must be called with
+// StringID returns the endpoint's ID in a string. Must be called with
 // the endpoint's mutex locked.
```

Comment:
We do not need to lock the mutex anymore with this function so this can be removed